Sources/xcodeproj/Objects/Project/PBXProject.swift

```diff
@@ -191,6 +191,44 @@ public final class PBXProject: PBXObject {
 
         return reference
     }
+    
+    
+    /// Adds a local swift package
+    ///
+    /// - Parameter path: Relative path to the swift package
+    /// - Parameter productName: The product to depend on
+    /// - Parameter target: Target to link package product to
+    /// - Parameter addFileReference: Include a file reference to the package (defaults to main group)
+    public func addLocalSwiftPackage(path: String,
+                                     productName: String,
+                                     target: PBXTarget? = nil,
+                                     addFileReference: Bool = true) -> XCSwiftPackageProductDependency {
+        let objects = try! self.objects()
+        
+        // Product
+        let productDependency = XCSwiftPackageProductDependency(productName: productName)
+        objects.add(object: productDependency)
+        target?.packageProductDependencies.append(productDependency)
+
+        // Build file
+        let buildFile = PBXBuildFile(product: productDependency)
+        objects.add(object: buildFile)
+        
+        // Link the product
+        try? target?.frameworksBuildPhase()?.files?.append(buildFile)
```

Comment:
I'd not swallow the errors thrown here.

Reply:
Moreover, if a frameworks build phase doesn't exist, this method doesn't do what's supposed to do. I'd rather throw an error and tell the user what's going on.

Reply:
OK, similarly should not be that the same for `addSwiftPackage`? Otherwise, agree, just tried to be consistent 🙂

Reply:
Yeah, it should be the same. I disregarded that when I implemented `addSwiftPackage`. My bad 😬